### ISSUE TITLE
Missing ASP.NET Core Runtime prompt does not download ASP.NET Core, but .NET Desktop instead

### ISSUE THREAD

**Issue by luislhg** [dotnet/sdk#33793]
### Description
I have a WPF application that is requiring Desktop Runtime and ASP.NET Core Runtime.
When the user does have Desktop runtime, but does not have ASP.NET Core runtime, windows will prompt to download it.
However, it ends up downloading the same .NET 6 Desktop and not the expected ASP.NET Core.
I'm not sure if it might be a bad url redirect in dotnet.microsoft...

### Configuration
Project main settings
```
<Project Sdk="Microsoft.NET.Sdk">
	<PropertyGroup>
		<OutputType>WinExe</OutputType>
		<TargetFramework>net6.0-windows</TargetFramework>
		<UseWPF>true</UseWPF>
		<UseWindowsForms>true</UseWindowsForms>
		<GenerateAssemblyInfo>false</GenerateAssemblyInfo>
	</PropertyGroup>
	<ItemGroup>
		<PackageReference Include="Microsoft.AspNetCore.SignalR.Client" Version="6.0.16" />
	</ItemGroup>
</Project>
```

### Other information
Video reproducing the issue: 
https://github.com/dotnet/core/assets/16278387/3caa0eef-2740-4755-a74a-e0155f95ec87

**Comment by vitek-karas** [comment]
@richlander @mairaw 
Another instance of wrong redirect. The URL the host navigates to is:
```
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.AspNetCore.App&framework_version=6.0.0&arch=x64&rid=win10-x64&gui=true
```

Which clearly states that it needs `Microsoft.AspNetCore.App` framework, but it redirects to the desktop runtime download instead.

**Comment by mairaw** [comment]
gui=true is the first check on the logic. I think folks indicated in the past that it was always a desktop download if that was true. We can fix that but it would be good to put a doc together on the service repo to clarify the behaviors @richlander 

It seems gui=true only indicates the request is coming from a desktop app.

**Comment by vitek-karas** [comment]
> It seems gui=true only indicates the request is coming from a desktop app.

Yes - that's exactly what it does. There's a sequencing problem such that if there's no .NET on the machine we may first construct a URL which says "We need Microsoft.NETCore.App" (because we can't even figure out what frameworks we will need without .NET on the machine) and when it's installed we would immediately fail with "Oh and now we need WindowsDesktop as well". So it makes sense to use the `gui=true` as a hint to offer the WindowsDesktop over base runtime.

So I would say the ideal behavior should be:
* No framework defined in the URL, or the framework is `Micosoft.NETCore.App`
  * If `gui=true` - redirect to the matching version of WindowsDesktop installer - we're guessing here a little bit, but it's an educate guess
  * If `gui=false` - redirect to core runtime download only
* Any other framework defined in the URL - redirect to that one, no matter what the `gui=true/false` says.

Note that the scenario in this case is not very common, it's an app which requires all 3 frameworks, core, WindowsDesktop and ASP.NET.

**Comment by mairaw** [comment]
There were some logic changes done recently to the service. I've tried fixing the behavior to attend this case:
https://github.com/dotnet/website-getdotnet/pull/114

**Comment by mairaw** [comment]
This should be working again. Thanks for reporting the issue!

**Comment by luislhg** [comment]
It is now downloading ASP.NET Core Runtime. Thanks!
However, it gets a different version and not the same one the user already has in the computer (.NET Core or .NET Desktop runtimes).

**Repro**
1) Computer has .NET Desktop Runtime 6.0.16 installed (consequently .NET Runtime 6.0.16 as well)
2) User tries to open a software that requires both .NET Desktop and ASP.NET Core.
3) The link downloads ASP.NET Core Runtime 6.2.0 (latest I assume) - This doesn't match the ones (Core and Desktop) he already has.
4) User tries to open the software again, it now complains about a missing .NET Core runtime and redirects to download .NET Desktop Runtime 6.0.20.

Now it works, but doesn't it sound too confusing and a lot of trouble for windows users?

**Comment by vitek-karas** [comment]
Thanks for reporting the second issue @luislhg . @mairaw we need to fix this - we need to offer download of an ASP.NET installer which chains the runtime installer as well. The download links should never offer ASP.NET without runtime - exactly due to the problem @luislhg reported, we need the matching runtime version. Same applies to WindowsDesktop which should also always come with the runtime.

**Comment by gmkado** [comment]
FWIW we ran into this recently, users don't realize the link is for a different install and think that something went wrong.

**Comment by marcpopMSFT** [comment]
Old issue triage: The original issue was fixed and looks like mairaw filed an issue to track the following up item. Closing as neither are .NET SDK issues.
